app/src/main/java/ch/epfl/polychef/recipe/SearchRecipe.java

```diff
@@ -82,10 +82,8 @@ private boolean compareName(String query, Recipe value) {
      */
 
     private boolean compareIngredient(String ingredient, Recipe value) {
-        String searchInput = ingredient;
-        searchInput = searchInput.toLowerCase();
         for (String ing : value.getIngredients().stream().map(Ingredient::getName).collect(Collectors.toList())) {
-            if (searchInput.contains(ing.toLowerCase()) || ing.toLowerCase().contains(searchInput)) {
+            if (Similarity.similarity(ingredient,ing)>0.3) {
```

Comment:
Define the 0.3 as a CONSTANT named attribute ?

Reply:
il faudrait (magic number), et aussi pour la maintenance